internal/controller/nodedisruption_controller.go

```diff
@@ -212,6 +218,20 @@ func (ndr *NodeDisruptionResolver) ValidateDisruption(ctx context.Context, budge
 	logger := log.FromContext(ctx)
 	any_failed = false
 
+	if nd.ImpactedNodes.Len() == 0 {
+		any_failed = true
+		status := nodedisruptionv1alpha1.DisruptedBudgetStatus{
```

Comment:
Can you add the Reference field but set it to the node disruption itself ?

Reply:
Reference expects a NamespacedName but ND is not namespaced

Reply:
NamespacedName.Namespace can be empty string (NodeDisruptionBudget are not namespaced as well)